site/src/pages/TemplatesPage/TemplatesPageView.tsx

```diff
@@ -114,94 +124,108 @@ export const TemplatesPageView: FC<React.PropsWithChildren<TemplatesPageViewProp
         )}
       </PageHeader>
 
-      <TableContainer>
-        <Table>
-          <TableHead>
-            <TableRow>
-              <TableCell width="50%">{Language.nameLabel}</TableCell>
-              <TableCell width="16%">{Language.usedByLabel}</TableCell>
-              <TableCell width="16%">{Language.lastUpdatedLabel}</TableCell>
-              <TableCell width="16%">{Language.createdByLabel}</TableCell>
-              <TableCell width="1%"></TableCell>
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            {props.loading && <TableLoader />}
-            {!props.loading && !props.templates?.length && (
+      {props.getOrganizationsError ? (
+        <ErrorSummary
+          error={props.getOrganizationsError}
+          defaultMessage={t("errors.getOrganizationsError")}
+        />
+      ) : props.getTemplatesError ? (
+        <ErrorSummary
+          error={props.getTemplatesError}
+          defaultMessage={t("errors.getTemplatesError")}
+        />
+      ) : (
+        <TableContainer>
```

Comment:
I find this triple, JSX ternary a little confusing and personally think this is a good candidate for early return.

Reply:
How would you do the early return? I'm not seeing a good way of doing it since the margins and page header need to be there in each case. I could put those in a const but at that point I'd rather put the content (error or data) in a const and still use the conditional so you can easily see what's definitely there vs what's conditional.

Reply:
I might move the margins and page header into `TemplatesPage.tsx` - they seem like page scaffolding to me. Then, one could decompose the `TemplatesPageView` file:
````
export const TemplatesPageView = () => {
  if (props.getOrganizationsError) {
    return <ErrorSummary />
  }

  if (props.getTemplatesError) {
    return <ErrorSummary />
  }

  if (empty) {
    return (
      <TableContainer>
        <LoadingTableBody />
      </TableContainer>
    )
  }

  return (
    <TableContainer>
      <TemplateTableBody templates={templates} />
    </TableContainer>
  )
}

````

I think the benefit here is we are writing state in a declarative way - it is apparent at a glance what states the developer should be aware of. Let me know what you think!

Reply:
Yeah, this looks nice and readable, but the scaffolding is in the view for storybook reasons. I'm open to a refactor but I think I'll leave it for future work so I can get the error handling in.

Reply:
sounds good!